### DISCUSSION TITLE
Runtime memory usage

### DISCUSSION THREAD

**Discussion by OmarMorceli** [zephyrproject-rtos/zephyr#51915] [Q&A]
I'm Using the nRF52840 DK and I'm facing a problem with my application, is there a way for me to monitor the RAM usage during runtime, so that I can determine whether the issue is due to memory usage or not ?

**Comment by henrikbrixandersen** [comment]
Have you tried using `CONFIG_THREAD_ANALYZER_AUTO=y` and friends?

**Comment by lidong1266** [comment]
do we have a shell command  (like Linux shell "free") to check the runtime memory usage on Zephyr?
